src/fastmcp/server/context.py

```diff
@@ -83,9 +84,20 @@ def my_tool(x: int, ctx: Context) -> str:
         request_id = ctx.request_id
         client_id = ctx.client_id
 
+        # Manage state across the request
+        ctx.state["user_id"] = "12345"
+        ctx.set_state_value("key", "value")
+        value = ctx.get_state_value("key")
+
         return str(x)
     ```
 
+    State Management:
+    Context objects maintain a state dictionary that can be used to store and share
+    data across middleware and tool calls within a request. When a new context
+    is created (nested contexts), it inherits a copy of its parent's state, ensuring
+    that modifications in child contexts don't affect parent contexts.
```

Comment:
Discussed the semantics of this a little in the PR description. If you think child state _should_ affect the parent, we could do it that way.